File: firewall_manager/app/schemas/object.py
from pydantic import BaseModel
from typing import List, Optional

class ObjectSearchRequest(BaseModel):
    device_ids: List[int]
    object_type: str  # 'network_object', 'network_group', 'service', 'service_group'

    # Common fields
    name: Optional[str] = None
    description: Optional[str] = None

    # NetworkObject specific
    ip_address: Optional[str] = None
    type: Optional[str] = None

    # Group specific
    members: Optional[str] = None # Member names, comma-separated

    # Service specific
    protocol: Optional[str] = None
    port: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/object.ts
export interface NetworkObject {
  id: number
  device_id: number
  name: string
  ip_address: string
  type: string
  description: string | null
  ip_version: string | null
  ip_start: string | null
  ip_end: string | null
  is_active: boolean
  last_seen_at: string | null
}

export interface NetworkGroup {
  id: number
  device_id: number
  name: string
  members: string
  description: string | null
  is_active: boolean
  last_seen_at: string | null
}

export interface Service {
  id: number
  device_id: number
  name: string
  protocol: string
  port: string
  description: string | null
  port_start: number | null
  port_end: number | null
  is_active: boolean
  last_seen_at: string | null
}

export interface ServiceGroup {
  id: number
  device_id: number
  name: string
  members: string
  description: string | null
  is_active: boolean
  last_seen_at: string | null
}

export interface ObjectSearchRequest {
  device_ids: number[]
  object_type: string
  name?: string
  description?: string
  ip_address?: string
  type?: string
  members?: string
  protocol?: string
  port?: string
  names?: string[]
  ip_addresses?: string[]
  protocols?: string[]
  ports?: string[]
  skip?: number
  limit?: number
}

export interface ObjectSearchResponse {
  network_objects: NetworkObject[]
  network_groups: NetworkGroup[]
  services: Service[]
  service_groups: ServiceGroup[]
}
